### PR TITLE
Refactoring in support of externally defined feature extractor libraries - i.e. LAISS

### PR DESCRIPTION
**Pull request by drewoldag**
This fairly substantial refactor removes hardcoded column names from `lightcurve_utils` and shifts the work over to the subclasses of `LightCurve`. The built in feature extractors (FEs) all use the same patterns, so we were able to keep the logic in the parent class.

This reduces the code in RESSPECT, and leaves the door open for external libraries that bring in a FE to implement their own method of providing feature column names.

### CODE REVIEW

File: src/resspect/database.py [hunk 1]
```diff
@@ -249,42 +255,15 @@ def load_features_from_file(self, path_to_features_file: str, screen=False,
...
-                                       'iotsu_lower_to_all_ratio', 'ilinear_fit_slope', 
-                                       'ilinear_fit_slope_sigma','ilinear_fit_reduced_chi2',
-                                       'zanderson_darling_normal','zinter_percentile_range_5',
-                                       'zchi2', 'zstetson_K', 'zweighted_mean','zduration', 
-                                       'zotsu_mean_diff','zotsu_std_lower', 'zotsu_std_upper',
-                                       'zotsu_lower_to_all_ratio', 'zlinear_fit_slope', 
-                                       'zlinear_fit_slope_sigma','zlinear_fit_reduced_chi2']
+        # Create the filter-feature names based on the survey.
+        survey_filters = FILTER_SETS[survey]
+        self.features_names = create_filter_feature_names(survey_filters, feature_names)
```
Comment (drewoldag) [inline src/resspect/database.py:264]: This is almost certainly insufficient. I believe that we need another method in the `LightCurve` base class that will allow us to return _just_ the feature names, excluding what is referred to as `metadata_names` here. 

Or more likely, we can break the methods into two, something like: 

```
def get_feature_header(...):
    return get_metadata_header() + get_features_header()

def get_metadata_header(...):
    # returns something like `['id', 'redshift', ...]

def get_features_header(...):
    # returns the cross product of features and filters
```

**Review by maxwest-uw — APPROVED** [review]

File: src/resspect/database.py [hunk 2]
```diff
@@ -234,6 +239,10 @@ def load_features_from_file(self, path_to_features_file: str, screen=False,
             "Bump", or "Malanchev". Default is "Bazin".
         """
 
+        if survey not in ['DES', 'LSST']:
```
Comment (drewoldag) [inline src/resspect/database.py:242]: @AmandaWasserman Is this logic correct, or should we support any `survey` that we find in the `FILTER_SETS` dictionary?  i.e. `DES`, `LSST`, and `SNPCC`

File: src/resspect/database.py (hunk 1 above)
Comment (drewoldag) [reply src/resspect/database.py]: Done

**Comment by AmandaWasserman** [comment]
Everything looks good doing a PR and run through of our typical operations for feature extraction, learn loop, and plotting

**Pull request merged**